### DISCUSSION TITLE
Scanning issue with qr

### DISCUSSION THREAD

**Discussion by RakeshBhaiKush** [aldinokemal/go-whatsapp-web-multidevice#143] [General]
The recent WhatsApp update has caused scanning issues for some users, particularly those using certain versions of the application. Users with the latest WhatsApp version are experiencing difficulty while scanning. also go through this threads 
(https://github.com/tulir/whatsmeow/issues/584)

**Comment by aldinokemal** [comment]
Answering late for anyone who lands here from search.

That `<stream:error code="516"/>` right after "Successfully paired" is server side, not a bug in gowa or in whatsmeow. WhatsApp rejects the freshly paired session based on the IP it connects from, which is why the same thread reported it working on a laptop but failing on a cloud VM, and why Baileys and whatsapp-web.js were hit at the same time. whatsmeow#584 was closed without a library fix for that reason, so upgrading alone will not change it.

What actually helps:

1. Pair on a normal connection, then move the session store to the server. The device store is `storages/whatsapp.db` by default (`src/config/settings.go:41`) and is overridable with `--db-uri` / `DB_URI` (`src/cmd/root.go:137`, `src/cmd/root.go:388`). Copy the whole `storages/` folder over and start the server, no rescan needed.

2. Send the WhatsApp websocket through a proxy. `WHATSAPP_PROXY` was added in `v8.11.0` and is applied with `SetProxyAddress` before connect (`src/infrastructure/whatsapp/init.go:108`, documented at `readme.md:249`):

```
WHATSAPP_PROXY=socks5://user:pass@host:1080
```

or `--whatsapp-proxy="socks5://user:pass@host:1080"` (`src/cmd/root.go:467`). Note that plain `HTTP_PROXY` does not work here, the whatsmeow ws dialer ignores it, which is why the dedicated setting exists.

3. If scanning is the awkward part rather than the IP, there is pairing by phone code since `v4.16.0`:

```
GET /app/login-with-code?phone=628912344551
```

(`src/ui/rest/app.go:21` -> `PairPhone` at `src/usecase/app.go:156`). It goes through the same connection though, so it is not a way around a blocked IP.

Also worth using a current release rather than a mid 2024 build, the whatsmeow dependency has moved a long way since then.
